packages/@n8n/ai-workflow-builder.ee/src/workflow-builder-agent.ts

```diff
@@ -320,53 +321,91 @@ export class WorkflowBuilderAgent {
 		const streamConfig = {
 			...threadConfig,
 			streamMode: ['updates', 'custom'],
-			recursionLimit: 30,
+			recursionLimit: 50,
 			signal: abortSignal,
 			callbacks: this.tracer ? [this.tracer] : undefined,
 		} as RunnableConfig;
 
-		const stream = await agent.stream(
-			{
-				messages: [new HumanMessage({ content: payload.message })],
-				workflowJSON: this.getDefaultWorkflowJSON(payload),
-				workflowOperations: [],
-				workflowContext: payload.workflowContext,
-			},
-			streamConfig,
-		);
-
 		try {
-			const streamProcessor = createStreamProcessor(stream);
-			for await (const output of streamProcessor) {
-				yield output;
+			const stream = await agent.stream(
+				{
+					messages: [new HumanMessage({ content: payload.message })],
+					workflowJSON: this.getDefaultWorkflowJSON(payload),
+					workflowOperations: [],
+					workflowContext: payload.workflowContext,
+				},
+				streamConfig,
+			);
+
+			try {
+				const streamProcessor = createStreamProcessor(stream);
+				for await (const output of streamProcessor) {
+					yield output;
+				}
+			} catch (error) {
```

Comment:
The nested try/catches with nested if/else-if are getting a bit hard to follow. Could we split it up to keep chat method lean? Something like:
```ts
async *chat(payload: ChatPayload, userId?: string, abortSignal?: AbortSignal) {
  // 1. Validate input
  this.validateMessage(payload.message);

  // 2. Setup configuration
  const { agent, streamConfig, threadConfig } = this.setupAgentConfiguration(
    payload, userId, abortSignal
  );

  // 3. Create and process stream
  try {
    const stream = await this.createStream(agent, payload, streamConfig);
    yield* this.processStream(stream, agent, threadConfig);
  } catch (error) {
    yield* this.handleStreamError(error, agent, threadConfig);
  }
}
```